File: PluginSource/Android/pushwoosh-unityplugin/src/main/java/com/pushwoosh/unityplugin/NotificationChannelDelegate.java
package com.pushwoosh.unityplugin;

public interface NotificationChannelDelegate {
    /**
     * @param channelName name of the channel specified in Android payload as "pw_channel" attribute.
     * If no attribute was specified, parameter gives default channel name
     * @return name that you want to assign to the channel on its creation. Note that empty name
     * will be ignored and default channel name will be assigned to the channel instead
     */
    String ChannelName(String channelName);
    /**
     * @param channelName name of the channel specified in Android payload as "pw_channel" attribute.
     * If no attribute was specified, parameter gives default channel name
     * @return description that you want to assign to the channel on its creation
     */
    String ChannelDescription(String channelName);
}

File: com.pushwoosh.unity.core/Runtime/NotificationChannelDelegate.cs
using System;
using UnityEngine;

public abstract class NotificationChannelDelegate : AndroidJavaProxy
{
    protected NotificationChannelDelegate() : base("com.pushwoosh.unityplugin.NotificationChannelDelegate") { }

    public abstract string ChannelName(string channelName);

    public abstract string ChannelDescription(string channelName);
}